### PR TITLE
Add DotNetPurple as a named color

### PR DESCRIPTION
**Pull request by maddymontaquila**
Sorry in advance for if I wasn't supposed to use tabs lol

### Description of Change

Added the official [dotnet brand color](https://github.com/dotnet/brand?tab=readme-ov-file#logo) - #512BD4 (0xFF512BD4) to Colors.cs file in Maui.Graphics

### Issues Fixed

My feelings

### CODE REVIEW

File: src/Graphics/src/Graphics/Colors.cs [hunk 1]
```diff
@@ -95,6 +95,8 @@ public static class Colors
 		public static readonly Color DimGrey = DimGray;
 		/// <summary>Gets the system-defined color that has an ARGB value of <c>#FF1E90FF</c>.</summary>
 		public static readonly Color DodgerBlue = Color.FromUint(0xFF1E90FF);
+		/// <summary>Gets the system-defined color that has an ARGB value of <c>#0xFF512BD4</c>.</summary>
+		public static readonly Color DotNetPurple = Color.FromUint(0xFF512BD4);
```
Comment (dotMorten) [inline src/Graphics/src/Graphics/Colors.cs:99]: Brand colors are not forever, so should probably be timestamped
```suggestion
		public static readonly Color DotNetPurple2024 = Color.FromUint(0xFF512BD4);
```

Comment (maddymontaquila) [reply src/Graphics/src/Graphics/Colors.cs]: this is why im not an engineer hehe

**Comment by davidortinau** [comment]
cc @terrajobst for the appropriate spelling/capitalization of the color name 👮‍♂️

**Comment by terrajobst** [comment]
Are the RBG values in those colors fixed/const?

If so, I'd argue that's a Bad Plan because branding changes. When I was your age (LOL), the purple was darker for example. And when .NET started it wasn't even purple, it was blue.

Framework Design Guideline: Do not use brand names in APIs.

**Comment by riverar** [comment]
Please avoid adding constants that unnecessarily expand the supported API surface. Let's focus on keeping the codebase streamlined and maintainable.

**Comment by riverar** [comment]
Consider code that may be accessing colors via index `Enum.GetValues(typeof(Color))[42]`.

It may be advisable to prepare a detailed specification for community review.

Meme: Zooming video shot of man trying desperately to hold his laugh

**Comment by dotMorten** [comment]
Taking a slide from one of Aaron Draplin's presentations:
image

**Comment by davidortinau** [comment]
@terrajobst fair review, thx. So what I'm understanding then is that as long as @maddymontaquila doesn't use a Brand name, then it's fair game. Like SassyPlum or SpicyThingThatIsPurple.

**Comment by Redth** [comment]
> @terrajobst fair review, thx. So what I'm understanding then is that as long as @maddymontaquila doesn't use a Brand name, then it's fair game. Like SassyPlum or SpicyThingThatIsPurple.

Do we need to make `XamarinBlue2011` into `YeOldHexagonBlue` ? Though, I would expect Xamarin branding is unlikely to change much in the future.

**Comment by maddymontaquila** [comment]
> @terrajobst fair review, thx. So what I'm understanding then is that as long as @maddymontaquila doesn't use a Brand name, then it's fair game. Like SassyPlum or SpicyThingThatIsPurple.

i MEAAAAAAN technically its a property not an API........... what about NotDotNetPurple also why are you no FUN!!! (in all seriousness though fair. what about DotPurple24 lol

**Comment by dotMorten** [comment]
@davidortinau no, [Sassy Plum](https://www.sassyplum.com/) is a brand name.

**Review by JonDouglas — APPROVED** [review]

**Comment by davidortinau** [comment]
> @davidortinau no, [Sassy Plum](https://www.sassyplum.com/) is a brand name.

image

File: src/Graphics/src/Graphics/Colors.cs [hunk 2]
```diff
@@ -95,6 +95,8 @@ public static class Colors
 		public static readonly Color DimGrey = DimGray;
 		/// <summary>Gets the system-defined color that has an ARGB value of <c>#FF1E90FF</c>.</summary>
 		public static readonly Color DodgerBlue = Color.FromUint(0xFF1E90FF);
+		/// <summary>Gets the system-defined color that has an ARGB value of <c>#0xFF512BD4</c>.</summary>
+		public static readonly Color DotNetPurple2024 = Color.FromUint(0xFF512BD4);
```
Comment (terrajobst) [inline src/Graphics/src/Graphics/Colors.cs:99]: Capitalization wise, that's consistent with priort art (e.g. `Microsoft.DotNet.*`). Adding the year is a way of future proofing, I guess.

**Comment by jassmith** [comment]
Failure to merge this would represent one of the greatest failures in software engineering history. Up there with NASA failing to convert metric and imperial units and smacking a lander into the Mars.

**Review by jfversluis — CHANGES_REQUESTED** [review]
Can you add a test please?

File: src/Graphics/src/Graphics/Colors.cs (hunk 2 above)
Comment (maddymontaquila) [reply src/Graphics/src/Graphics/Colors.cs:99]: good thank you

**Comment by RobbiewOnline** [comment]
It's for colour aliases only, don't pollute with any branding - for any company 🙈

if it's desired, suggest it the the community toolkit for consideration.

**Comment by bitzeal-johan** [comment]
Please focus on adding things that are useful to the .net community as a whole. Not things that are fun and gimmicky to microsoft employees

**Comment by Vroomer** [comment]
I'm so glad that all critical bugs have been fixed, layouts are no longer going bonkers with measurments and CollectionViews are flying like rockets, so that you can finally get to what really matters, i.e. triggering people by adding some much requested fancy brand colors! In all seriousness tho, I vote for SassyPlum! <3

**Comment by bcaceiro** [comment]
So kewl. Now make a complex app using this new brand color and publish it :) not just a hello world!

**Comment by bitzeal-johan** [comment]
> > Please focus on adding things that are useful to the .net community as a whole. Not things that are fun and gimmicky to microsoft employees
> 
> Please focus on creating a new PR in this repo[1](#user-content-fn-1-dc966f74962768de06aba79476200b96) so that I (read everybody) can benefit ;) Thank you.
> 
> ## Footnotes
> 1. I don't see any from you just yet. [↩](#user-content-fnref-1-dc966f74962768de06aba79476200b96)

At least I'm not making things worse :)

**Comment by davidortinau** [comment]
@maddymontaquila it might be best if you Fork the repo in order to enjoy this change.

**Comment by maddymontaquila** [comment]
Well, now that I know who the fun people are, I'll close this so the drama can end 🤣 Maybe someone will sneak it into the template for me instead 😬 I made a NuGet though to spite the rest of you 😊 https://www.nuget.org/packages/FunColors/ 

It took me less time to make this PR and subsequent NuGet than it took me to read these comments so fear not, I'm not distracting from "real" product work 😉 #PurpleGate has only just begun!!!!!!

**Comment by kazo0** [comment]
[Petition to reinstate this PR](https://chng.it/QT4RMZ2mQp)

#JusticeForMaddy #PurpleGate

**Comment by dersia** [comment]
it's sad to see this PR geing closed and seeing some of the comments in this PR makes me feel frustrated about our community.

sure there are important other issues to fix in Maui, but I see this as evenly important, because someone from our community has a desire that they'd like to bring in as a contribution. and as long as there is nothing technically wrong with this PR it should be accepted.

As for brand names for color names. for me the big difference is, that the contributor is not trying to bring in a brand name on an API or Property, but they are bringing in a Color name. and there are multiple examples for Color names that include a brand/company. in Germany we have the Telekom-Magenta color and there are others like that out there.

so for what it's worth, I think this Contribution is not just about @maddymontaquila feelings, as stated in the issue description, but a normal contribution that a named color is added that is missing.

I support this, because I hat to Google hexcode for known colors, we should really add other named colors too.

**Pull request closed without merging**